core/src/main/scala/cilib/Entity.scala

```diff
@@ -117,6 +141,19 @@ object Position {
 
   def createCollection[A](f: Position[Double] => A)(domain: NonEmptyList[Interval[Double]], n: Int): RVar[List[A]] =
     createPositions(domain,n).map(_.map(f))
+
+  def mean[A](positions: NonEmptyList[Position[A]])(implicit R: Field[A]) =
+    (R.one / positions.size.toDouble) *: positions.list.toList.reduce(_ + _)
+
```

Comment:
Can we remove the `.list.toList`` here as a NonEmptyList will always have 1 or more entries?